### DISCUSSION TITLE
Crash in vkGetPhysicalDeviceSurfaceCapabilitiesKHR, no validation messages

### DISCUSSION THREAD

**Discussion by rnikander** [KhronosGroup/MoltenVK#2171] [Q&A] [answered]
I'm just getting started with this, getting "hello triangle" to deploy and run on various devices. On macOS it works, but on an iOS device, I get part way through initialization, but then Xcode reports: `Thread 1: EXC_BAD_ACCESS (code=1, address=0x41a1f622da11)` when I call:

```
vkGetPhysicalDeviceSurfaceCapabilitiesKHR(device, m_surface, &details.capabilities);
```

I can see from the stack it's somewhere in...

```
MVKPhysicalDevice::getSurfaceCapabilities(VkPhysicalDeviceSurfaceInfo2KHR const*, VkSurfaceCapabilities2KHR*)
```

But it shows me only machine code. 

I guess the next step could be to build MoltenVK so I have a debuggable version of the library and can see what line it's crashing on? 

Any other suggestions?

**Comment by billhollings** [comment]
I'm not able to replicate this with the Cube demo in MoltenVK. And the code for `vkGetPhysicalDeviceSurfaceCapabilitiesKHR()` is the same for macOS and iOS, so it should behave similar on both platforms.

I assume it must have something to do with the arguments you're passing into the call. Make sure required items are not `nullptr`.

If you are still having issues, yes, you can build a debug version of MoltenVK to get a full stack trace and report it back here.

**Reply by rnikander** [reply]
I copied some old sample code and I think I'm using the wrong extension. Let me try the "metal" instead of "iOS" one.

**Reply by rnikander** [reply, marked as answer]
I changed from calling `vkCreateIOSSurfaceMVK` to `vkCreateMetalSurfaceEXT`, and now it works. If the first one is obsolete/dead (?), maybe it should be issuing warnings and not returning VK_SUCCESS?
